pkg/pgmodel/ingestor/metric_batcher_test.go

```diff
@@ -0,0 +1,78 @@
+package ingestor
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/timescale/promscale/pkg/pgmodel/model"
+)
+
+func TestMetricTableName(t *testing.T) {
+	testCases := []struct {
+		name        string
+		tableName   string
+		errExpected bool
+		sqlQueries  []model.SqlQuery
+	}{
+		{
+			name:      "no error",
+			tableName: "res1",
+			sqlQueries: []model.SqlQuery{
+				{
+					Sql:     "SELECT table_name, possibly_new FROM _prom_catalog.get_or_create_metric_table_name($1)",
+					Args:    []interface{}{"t1"},
+					Results: model.RowResults{{"res1", true}},
+				},
+			},
+		},
+		{
+			name:      "no error2",
+			tableName: "res2",
+			sqlQueries: []model.SqlQuery{
+				{
+					Sql:     "SELECT table_name, possibly_new FROM _prom_catalog.get_or_create_metric_table_name($1)",
+					Args:    []interface{}{"t1"},
+					Results: model.RowResults{{"res2", true}},
+				},
+			},
+		},
+		{
+			name:        "error",
+			tableName:   "res1",
+			errExpected: true,
+			sqlQueries: []model.SqlQuery{
+				{
+					Sql:  "SELECT table_name, possibly_new FROM _prom_catalog.get_or_create_metric_table_name($1)",
+					Args: []interface{}{"t1"},
+					Err:  fmt.Errorf("test"),
+				},
+			},
+		},
+		{
+			name:        "empty table name",
+			tableName:   "res2",
+			errExpected: true,
+			sqlQueries: []model.SqlQuery{
+				{
+					Sql:     "SELECT table_name, possibly_new FROM _prom_catalog.get_or_create_metric_table_name($1)",
+					Args:    []interface{}{"t1"},
```

Comment:
I do not understand sql recorder perfectly, but shouldn't this be emty string?

Reply:
No this is the argument - the metric name - not the table name. The response table name is on the next line and it's empty